thamos/lib.py

```diff
@@ -115,6 +140,10 @@ def _no_spinner():
         while True:
             try:
                 response = status_func(analysis_id)
+            except TimeoutError as exc:
```

Comment:
What about something like this:

```python
start_time = time.monotonic()
while True:
 # ...
 if _THAMOS_TIMEOUT and time.monotonic() - start_time > _THAMOS_TIMEOUT:
   raise TimeoutError("Thoth backend did not respond in time, timeout set to {_THAMOS_TIME}")
 # ...
```

The think is that thamos can be used as a library. We don't know how users will use Thamos in their applications and calling ``time.alarm`` discards any previously scheduled alarm that can be setup by user. Also consider maintainability of the code.

Reply:
The timer won't stop the execution of `status_func` isn't that the purpose?
```
    with spinner():
        sleep(2)  # TODO: remove once we fully run on Argo workflows
        start_time = time.monotonic()
        while True:
            if _THAMOS_TIMEOUT and time.monotonic() - start_time > _THAMOS_TIMEOUT:
                raise TimeoutError("Thoth backend did not respond in time, timeout set to {_THAMOS_TIME}")
            try:
                sleep(1000)
                response = status_func(analysis_id)
            except TimeoutError as exc:
                _LOGGER.error(
                    "Failed to obtain status from Thoth without timeout limit: %s", str(exc))
                raise
```

Reply:
```
                sleep(1000)
```

This will not happen - we have control over how much time is spent in `status_func` or in `_wait_for_analysis`. In other words, we are preventing how many result-fetch loops is performed given the timeout. I think the implementation you provided in #423 is correct.

Reply:
> This will not happen - we have control over how much time is spent in `status_func` or in `_wait_for_analysis`. 
How do we do we control how much time is spent in `status_func`, just curious? Thanks

Reply:
It's out of our control. The underlying library used implements HTTP timeout and raises if the connection time outs. We are controlling timeouts on application layer timeout that Thoth did not compute results given time allocated.

Reply:
Thanks that makes sense.